src/libraries/Common/src/Roslyn/MetadataHelpers.cs

```diff
@@ -0,0 +1,1034 @@
+
+                        int cmp = nameComparer.Compare(lastChildNamespaceName, childNamespaceName);
+                        if (cmp == 0)
+                        {
+                            // We are still processing the same child namespace
+                            typesInLastChildNamespace.Add(pair);
+                        }
+                        else
+                        {
+                            // This is a new child namespace
+                            if (cmp > 0)
+                            {
+                                // The sort order is violated for child namespace names. Obfuscation is the likely reason for this. 
+                                Debug.Assert((object)lastChildNamespaceName != null);
+                                possiblyHavePairsWithDuplicateKey = true;
+                            }
+
+                            // Preserve information about previous child namespace.
+                            if (typesInLastChildNamespace != null)
+                            {
+                                Debug.Assert(typesInLastChildNamespace.Count != 0);
+                                nestedNamespaces.Add(
+                                    new KeyValuePair<string, IEnumerable<IGrouping<string, TypeDefinitionHandle>>>(
+                                        lastChildNamespaceName, typesInLastChildNamespace));
+                            }
+
+                            typesInLastChildNamespace = new List<IGrouping<string, TypeDefinitionHandle>>();
+                            lastChildNamespaceName = childNamespaceName;
+                            Debug.Assert((object)lastChildNamespaceName != null);
+
+                            typesInLastChildNamespace.Add(pair);
+                        }
+                    }
+                    while (enumerator.MoveNext());
+
+                    // Preserve information about last child namespace.
+                    if (typesInLastChildNamespace != null)
+                    {
+                        Debug.Assert(typesInLastChildNamespace.Count != 0);
+                        nestedNamespaces.Add(
+                            new KeyValuePair<string, IEnumerable<IGrouping<string, TypeDefinitionHandle>>>(
+                                lastChildNamespaceName, typesInLastChildNamespace));
+                    }
+
+DoneWithSequence:
+/*empty statement*/
+                    ;
+                }
+            } // using
+
+            types = nestedTypes;
+
+            // Merge pairs with the same key
+            if (possiblyHavePairsWithDuplicateKey)
+            {
+                var names = new Dictionary<string, int>(nestedNamespaces.Count, nameComparer);
+
+                for (int i = nestedNamespaces.Count - 1; i >= 0; i--)
+                {
+                    names[nestedNamespaces[i].Key] = i;
+                }
```

Comment:
Why iterate in reverse to add everything to the dictionary?

Reply:
@jaredpar for feedback on this existing roslyn code :)

Reply:
actually, i can remove *all* this code (At the cost of more deviations from roslyn).  you guys don't ever look for nested-attribute types (e.g. the attribute is a nested type, not that hte attribute is on a nested type).  So i can just remove this and hardcode that knowlege.